src/Microsoft.VisualStudio.Threading/ReentrantSemaphore.cs

```diff
@@ -167,7 +168,37 @@ public static ReentrantSemaphore Create(int initialCount = 1, JoinableTaskContex
     /// </param>
     /// <param name="cancellationToken">A cancellation token.</param>
     /// <returns>A task that completes with the result of <paramref name="operation"/>, after the semaphore has been exited.</returns>
-    public abstract ValueTask<T> ExecuteAsync<T>(Func<ValueTask<T>> operation, CancellationToken cancellationToken = default);
+    public ValueTask<T> ExecuteAsync<T>(Func<ValueTask<T>> operation, CancellationToken cancellationToken = default)
+        => this.ExecuteValueTaskAsync<T>(operation, null, cancellationToken);
+
+    /// <summary>
+    /// Executes a given operation within the semaphore.
+    /// </summary>
+    /// <param name="operation">
+    /// The delegate to invoke once the semaphore is entered. If a <see cref="JoinableTaskContext"/> was supplied to the constructor,
+    /// this delegate will execute on the main thread if this is invoked on the main thread, otherwise it will be invoked on the
+    /// threadpool. When no <see cref="JoinableTaskContext"/> is supplied to the constructor, this delegate will execute on the
+    /// caller's context.
+    /// </param>
+    /// <param name="cancellationToken">A cancellation token.</param>
+    /// <returns>A task that completes with the result of <paramref name="operation"/>, after the semaphore has been exited.</returns>
+    public Task ExecuteAsync(Func<object?, Task> operation, object? state, CancellationToken cancellationToken = default)
```

Comment:
`state` params are missing doc nodes in these new overloads.

Reply:
Ya, and a draft that *may* never be completed.